test/common/network/addr_family_aware_socket_option_impl_test.cc

```diff
@@ -168,7 +174,9 @@ TEST_F(AddrFamilyAwareSocketOptionImplTest, GetSocketOptionCannotDetermineVersio
                                                 Network::SocketOptionName(std::make_pair(6, 11)),
                                                 5};
 
-  EXPECT_CALL(socket_, fd()).WillOnce(Return(-1));
+  IoHandlePtr io_handle = std::make_unique<IoSocketHandle>();
+  ScopedIoHandleCloser closer(io_handle);
```

Comment:
Do we need this closer? |io_handle| created above is not associated with any socket (fd = -1), if the io_handle->close() implements closing socket in the following PR, close(-1) will cause problem.

Reply:
We need this exactly because the io_handle is not associated with the socket, and so the socket destructor is not going to close the io_handle- the assert in IoSocketHandle destructor will go off. As for a close of (-1), the code can implement the close only if the fd != -1.

Reply:
I agree. I commented it out and the test didn't fail, so I was confusing it with another closer in the same file.